src/libPMacc/include/random/distributions/uniform/Uniform_float.hpp

```diff
@@ -48,10 +54,125 @@ namespace detail
         DINLINE result_type
         operator()(StateType& state)
         {
-            return _curand_uniform(RNGMethod().get32Bits(state));
+            const result_type 2pow32Inv = 2.3283064e-10f;
```

Comment:
I think we should use `float` here instead of result_type as the constants are `float`